src/main/java/com/squareup/kotlinpoet/CodeBlock.kt

```diff
@@ -50,13 +50,65 @@ import kotlin.reflect.KClass
  *    double-indented.
  *  * `%]` ends a statement.
  */
-class CodeBlock private constructor(builder: CodeBlock.Builder) {
+class CodeBlock private constructor(
+    internal val formatParts: List<String>,
+    internal val args: List<Any?>) {
   /** A heterogeneous list containing string literals and value placeholders.  */
-  internal val formatParts: List<String> = builder.formatParts.toImmutableList()
-  internal val args: List<Any?> = builder.args.toImmutableList()
 
   fun isEmpty() = formatParts.isEmpty()
 
+  /**
+   * Returns a code block with `prefix` stripped off, or null if this code block doesn't start with
+   * `prefix`.
+   *
+   * This is a pretty basic implementation that might not cover cases like mismatched whitespace. We
+   * could offer something more lenient if necessary.
+   */
+  internal fun withoutPrefix(prefix: CodeBlock): CodeBlock? {
+    if (formatParts.size < prefix.formatParts.size) return null
+    if (args.size < prefix.args.size) return null
+
+    var prefixArgCount = 0
+    var firstFormatPart: String? = null
+
+    // Walk through the formatParts of prefix to confirm that it's a of this.
+    prefix.formatParts.forEachIndexed { index, formatPart ->
+      if (formatParts[index] != formatPart) {
+        // We've found a format part that doesn't match. If this is the very last format part check
+        // for a string prefix match. If that doesn't match, we're done.
+        if (index == prefix.formatParts.size - 1 && formatParts[index].startsWith(formatPart)) {
+          firstFormatPart = formatParts[index].substring(formatPart.length)
+        } else {
+          return null
+        }
+      }
+
+      // If the matching format part has an argument, check that too.
+      if (formatPart.startsWith("%") && !isNoArgPlaceholder(formatPart[1])) {
+        if (args[prefixArgCount] != prefix.args[prefixArgCount]) {
+          return null // Argument doesn't match.
+        }
+        prefixArgCount++
+      }
+    }
+
+    // We found a prefix. Prepare the suffix as a result.
+    val resultFormatParts = ArrayList<String>()
+    if (firstFormatPart != null) {
+      resultFormatParts.add(firstFormatPart as String)
```

Comment:
It's because it can't guarantee the second read will be non-null.

Try
```
firstFormatPart?.let {
  resultFormatPart.add(it)
}
```
to only read the var once.